### PR TITLE
dependency graph filtering

### PR DESCRIPTION
**Pull request by bulicekj**
Changing the dropdown select on the dependency graph filtering to recreate the graph with one level up and down from the selected service (#2535)

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
thanks! gotta screen shot?

**Comment by codefromthecrypt** [comment]
fyi this is likely delaying review due to "golden week" in japan

**Comment by tacigar** [comment]
Sorry for delaying review 😭 
Thank you for your great work!

**Comment by tacigar** [comment]
Linter has issued a warning.
Could you fix them?

**Comment by tacigar** [comment]
There are some lint errors.
Would you fix them? 🙇

File: zipkin-lens/src/components/Dependencies/DependenciesGraph.js [hunk 1]
```diff
@@ -71,8 +79,8 @@ class DependenciesGraph extends React.Component {
             name: 'dependencies-graph',
             updated: new Date().getTime(),
             maxVolume: maxVolume * 2000,
-            nodes: graph.allNodes(),
-            connections: graph.allEdges(),
+            nodes,
+            connections
           }}
           objectHighlighted={this.handleObjectHighlighted}
           match={filter}
```
Comment (tacigar) [inline zipkin-lens/src/components/Dependencies/DependenciesGraph.js]: `match` prop is no longer needed.
Would you remove this line?

**Comment by bulicekj** [comment]
Before: 
Screen Shot 2019-04-29 at 11 09 02 AM
After:
Screen Shot 2019-04-29 at 11 16 36 AM


FYI - there's an bug with Vizceral (the dependency graph framework) stacking nodes when you switch to a service that isn't present on the graph. For example, selecting `frontend` after previously selecting `frontendtwo` like the above screenshot will cause this to happen: 
Screen Shot 2019-04-29 at 11 15 32 AM

This should probably not be merged until we can figure out how to prevent nodes from stacking.

File: zipkin-lens/src/components/Dependencies/DependenciesGraph.js (hunk 1 above)
Comment (bulicekj) [reply zipkin-lens/src/components/Dependencies/DependenciesGraph.js]: Good call 👍

**Review by tacigar — APPROVED** [review]

**Comment by tacigar** [comment]
Thank you so much!
LGTM 👍 
But we need to investigate the above issue a little more before merging

**Comment by codefromthecrypt** [comment]
any news on the stacking front (or lack of stacking front)?

**Comment by bulicekj** [comment]
I spent some time trying to change the way the graph is re-rendered upon the state change, but no success thus far. @tacigar any ideas?

**Comment by tacigar** [comment]
Ummm...
I tried to re-render, but I could not do...
Maybe it is better to use **Vizceral** directly instead of **Vizceral-react**.
If do so, we can handle more.

**Comment by codefromthecrypt** [comment]
naver pinpoint has an upstream and downstream depth control to their filter. for example, if you select a service you can show one link up and down by default, but you can change that to show 2 links down. I don't think their data model is precise enough to truly filter that the downstream were caused by the selected service though. I think it doesn't

**Review by zeagord — APPROVED** [review]

**Comment by bulicekj** [comment]
@tacigar @adriancole I was playing around today and found a solution that fixes the above mentioned bug. If we completely un-mount and re-mount VizceralExt, it re-renders with the correct layout -> 
zipkin-vizceral-2 

It's not the most elegant solution, as it hacks thorough some of the react re-rendering magic, but fixes the "stacking" issue. Let me know what you think

**Comment by bulicekj** [comment]
Also worth a note, I tried playing around with [vizceral's built in filtering](https://github.com/Netflix/Vizceral/wiki/Configuration#filters) and got it to work with one state change but couldn't get it to filter again after, even though [state should update](https://github.com/Netflix/vizceral-react/blob/b40765c9e22cba4b363e74ead011b578a8e65935/src/vizceral.jsx#L77) 🤔

**Comment by tacigar** [comment]
Wow...wonderful hack...

**Review by tacigar — APPROVED** [review]

**Comment by tacigar** [comment]
Thank you for your great work!!

**Comment by codefromthecrypt** [comment]
@tacigar which merges first? this or the other?

**Comment by tacigar** [comment]
Please merge this first!
I think @bulicekj 's great work should be on the commit history!

**Pull request merged**